### PR TITLE
Zhouyuan/edge mesh input

### PR DESCRIPTION
**Pull request by Zhouyuan-Chen**
Reopen the edge mesh input. I closed this PR since I changed the name recently.

This part's EdgeMesh is outdated and need to update that part code.

### CODE REVIEW

File: src/wmtk/io/EdgeMeshReader.hpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#include <Eigen/Core>
+#include <filesystem>
+#include <fstream>
+
+namespace wmtk {
+class Mesh;
+
+class EdgeMeshReader
+{
+public:
```
Comment (daniel-zint) [inline src/wmtk/io/EdgeMeshReader.hpp:10]: Make sure that only those functions are `public` that need to be. The rest should be `private`.

File: src/wmtk/io/EdgeMeshReader.hpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    enum data_type { V, L, VT, VN, VP, COMMENT };
+    enum file_type { OBJ, OFF };
+    EdgeMeshReader(const std::string& filename, const file_type type);
+    void read(
```
Comment (daniel-zint) [inline src/wmtk/io/EdgeMeshReader.hpp:14]: All these functions need to covered in tests.

File: components/wmtk_components/input/input.cpp [hunk 3]
```diff
@@ -52,11 +56,65 @@ void input(const nlohmann::json& j, std::map<std::string, std::filesystem::path>
         files[options.name] = cached_mesh_file;
         break;
     }
-    case 1:
+    case 1: {
         // edge mesh
-        throw "not implemented";
-        assert(false);
+        spdlog::warn("this edge mesh is old version, need to update according to the latest "
```
Comment (daniel-zint) [inline components/wmtk_components/input/input.cpp:61]: Now as you have the `EdgeMesh` available, you should remove the warning and adapt the code.

**Review by daniel-zint — CHANGES_REQUESTED** [review]
The new classes requires testing. I assume the code works but we need unit tests to prove it.

**Comment by Zhouyuan-Chen** [comment]
Added the tests.
Edge Mesh Input finished.

**Comment by daniel-zint** [comment]
Whenever you add the tag `[.]` to a test, it is ignored. You need to remove them.

**Review by daniel-zint — CHANGES_REQUESTED** [review]
I'll review it once the tests are running in the CLI.

You can upload your OBJ file for testing here:
https://github.com/wildmeshing/data

**Comment by Zhouyuan-Chen** [comment]
This component will not be used anymore, I will delete this branch later.

**Pull request closed without merging**